packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,274 @@
+    /**
+     * Optional; indicate if inking on the selected Image is allowed or not
+     * Default value is true
+     */
+    ink?: boolean;
+
+    /**
+     * Optional; indicate if putting text stickers on the selected Image is allowed or not
+     * Default value is true
+     */
+    textSticker?: boolean;
+
+    /**
+     * Optional; indicate if image filtering mode is enabled on the selected image
+     * Default value is false
+     */
+    enableFilter?: boolean;
+
+    /**
+     * Optional; Lets the developer specify the image output formats, more than one can be specified.
+     * Default value is Image.
+     */
+    imageOutputFormats?: ImageOutputFormats[];
+  }
+
+  /**
+   * The modes in which camera can be launched
+   * @beta
+   */
+  export enum CameraStartMode {
+    /** Photo mode. */
+    Photo = 1,
+    /** Document mode. */
+    Document = 2,
+    /** Whiteboard mode. */
+    Whiteboard = 3,
+    /** Business card mode. */
+    BusinessCard = 4,
+  }
+  /**
+   * Enum for file formats supported
+   * @beta
+   */
+  export enum FileFormat {
+    /** Base64 encoding */
+    Base64 = 'base64',
+    /** File id */
+    ID = 'id',
+  }
+  /**
+   * Specifies the image source
+   * @beta
+   */
+  export enum Source {
+    /** visual media source is camera. */
+    Camera = 1,
+    /** visual media source is gallery. */
+    Gallery = 2,
+  }
+  /**
+   * Specifies the image output formats.
+   * @beta
+   */
+  export enum ImageOutputFormats {
+    /** Outputs image.  */
+    IMAGE = 1,
+    /** Outputs pdf. */
+    PDF = 2,
+  }
+
+  /**
+   * VisualMediaFile object that can be used to represent image or video
+   *
+   * @beta
+   */
+  export class VisualMediaFile {
+    /**
+     * Content of the file. When format is Base64, this is the base64 content
+     * When format is ID, this is id mapping to the URI
+     * When format is base64 and app needs to use this directly in HTML tags, it should convert this to dataUrl.
+     */
+    public content: string;
+
+    /**
+     * Format of the content
+     */
+    public format: FileFormat;
+
+    /**
+     * Size of the file in KB
+     */
+    public size: number;
+
+    /**
+     * MIME type. This can be used for constructing a dataUrl, if needed.
+     */
+    public mimeType: string;
+
+    /**
+     * Optional: Name of the file
+     */
+    public name?: string;
+    /**
+     * A preview of the file which is a lightweight representation.
+     * In case of images this will be a thumbnail/compressed image in base64 encoding.
+     */
+    public preview: string;
+  }
+
+  /**
+   * Checks whether or not visualMedia has user permission
+   *
+   * @beta
+   * @returns Promise that will resolve with true if the user had granted the app permission to media information, or with false otherwise,
+   * In case of an error, promise will reject with the error. Function can also throw a NOT_SUPPORTED_ON_PLATFORM error
```

Comment:
>Function can also throw a NOT_SUPPORTED_ON_PLATFORM error [](http://example.com/codeflow?start=62&length=57)

Can you use the [@throws]([https://typedoc.org/tags/throws/#:~:text=The%20%40throws%20tag%20can%20be%20used%20to%20document,can%20be%20thrown%20by%20a%20function%20or%20method](https://typedoc.org/tags/throws/#:~:text=The%20%40throws%20tag%20can%20be%20used%20to%20document,can%20be%20thrown%20by%20a%20function%20or%20method).) tag to document this?